unified-installer/astra-unified-installer.sh

```diff
@@ -1117,12 +1179,177 @@ is_astra_registry() {
+    local -ar registries=("$@")
+    for reg in "${registries[@]}" ; do
+        if is_astra_registry "$reg"; then
+            echo "$reg"
+            return 0
+        fi
+    done
+    return 1
+}
+
+find_astra_registry_in_connector_images() {
+    find_astra_registry "$CONNECTOR_IMAGE_REGISTRY" "$NEPTUNE_IMAGE_REGISTRY" "$CONNECTOR_OPERATOR_IMAGE_REGISTRY"
+}
+
+find_astra_registry_in_trident_images() {
+    find_astra_registry "$TRIDENT_OPERATOR_IMAGE_REGISTRY" "$TRIDENT_AUTOSUPPORT_IMAGE_REGISTRY" \
+        "$TRIDENT_IMAGE_REGISTRY" "$TRIDENT_ACP_IMAGE_REGISTRY"
+}
+
+generate_kustomization_dirs_and_files() {
+    local -r kustomization_file="$1"
+    local -r add_as_resource_to_top_kustomization_file="${2:-"false"}"
+    local -r kustomization_dir="$(dirname "$kustomization_file")"
+
+    mkdir -p "$kustomization_dir"
+    if [ ! -f "$kustomization_file" ]; then
+        cat <<EOF > "$kustomization_file"
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+patches:
+transformers:
+generatorOptions:
+  disableNameSuffixHash: true
+secretGenerator:
+EOF
+        logdebug "$kustomization_file: OK"
+    fi
+
+    if [ "$add_as_resource_to_top_kustomization_file" == "true" ]; then
+        local -r top_kustomization_file="$__GENERATED_KUSTOMIZATION_FILE"
+        local -r base_dir="$(basename "$kustomization_dir")"
+        insert_into_file_after_pattern "$top_kustomization_file" "resources:" "- ./$base_dir"
+        logdebug "$top_kustomization_file: added resource entry for resource at path '$base_dir'"
+    fi
+}
+
+add_astra_registry_secret_to_kustomization() {
+    local -r kustomization_file="${1}"
+    local -r namespace="${2}"
+    local -r registry="${3}"
+    [ -z "$kustomization_file" ] && fatal "no kustomization file given"
+    [ ! -f "$kustomization_file" ] && fatal "kustomization file '$kustomization_file' does not exist"
+    [ -z "$namespace" ] && fatal "no namespace given"
+    [ -z "$registry" ] && fatal "no registry given"
+
+    local -r account_id="$ASTRA_ACCOUNT_ID"
+    local -r api_token="$ASTRA_API_TOKEN"
+    local -r pull_secret="$__ASTRA_REGISTRY_IMAGE_PULL_SECRET"
+    [ -z "$account_id" ] && fatal "no account_id found"
+    [ -z "$api_token" ] && fatal "no api_token found"
+    [ -z "$pull_secret" ] && fatal "no pull_secret found"
+
+    local -r encoded_creds=$(echo -n "$account_id:$api_token" | base64)
+
+    # Add secret generator
+    insert_into_file_after_pattern "$kustomization_file" "secretGenerator:" \
+'- name: "'"$pull_secret"'"
+  namespace: "'"$namespace"'"
+  type: kubernetes.io/dockerconfigjson
+  literals:
+  - |
+    .dockerconfigjson={
+      "auths": {
+        "'"$registry"'": {
+          "username": "'"$account_id"'",
+          "password": "'"$api_token"'",
+          "auth": "'"$encoded_creds"'"
+        }
+      }
+    }'
+    logdebug "$kustomization_file: added secret '$pull_secret' to namespace '$namespace'"
+}
+
+add_api_token_secret_to_kustomization() {
```

Comment:
This is the other part of the step that `add_astra_registry_secret_to_kustomization` was also a part of. This one is only used for the connector, but figured I'd make it a generic function and place it next to the other since they're pretty similar.